machine/v1beta1/types_awsplacementgroup.go

```diff
@@ -0,0 +1,164 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AWSPlacementGroup ensures that a placement group matching the given configuration exists within AWS
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".spec.groupType",description="Placement Group Type"
+// +kubebuilder:printcolumn:name="Management",type="string",JSONPath=".spec.managementState",description="Management State"
+// +kubebuilder:printcolumn:name="Replicas",type="integer",JSONPath=".status.replicas",description="EC2 Replicas within the Placement Group"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="AWSPlacementGroup age"
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type AWSPlacementGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AWSPlacementGroupSpec   `json:"spec,omitempty"`
+	Status AWSPlacementGroupStatus `json:"status,omitempty"`
+}
+
+type AWSPlacementGroupSpec struct {
+	// ManagementState determines whether the placement group is expected
+	// to be managed by this CRD or whether it is user managed.
+	// A managed placement group may be moved to unmanaged, however an unmanaged
+	// group may not be moved back to managed.
+	// +kubebuilder:default:=Managed
+	// +optional
+	ManagementState ManagementState `json:"managementState"`
+
+	// CredentialsSecret is a reference to the secret with AWS credentials. Otherwise, defaults to permissions
+	// provided by attached IAM role where the actuator is running.
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
```

Comment:
don't reuse the corev1.*Reference types.  Instead, write your own SecretReference with the fields you want.  Depending on external types makes them change you don't notice.  I wrote something up for kube a while back.

Reply:
Thanks for the link, I'm convinced and that's a good reference for why I created this in case anyone asks me in the future, I've updated to create our own type already, so current code should be updated